packages/babel-plugin-transform-built-ins/src/builtins.js

```diff
@@ -0,0 +1,79 @@
+module.exports = {
```

Comment:
The way babel does this is to maintain only `[String, Number, Math]` 

https://github.com/babel/babel/blob/f8ffe03/packages/babel-traverse/src/path/evaluation.js#L5 

So, I think it's relatively safe to assume that `Math.x` will be a side-effect free evaluation except `Math.random`